### PR TITLE
[GraphBolt] add pydantic-based metadata for TVT

### PR DESCRIPTION
**Pull request by Rhett-Ying**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: python/dgl/graphbolt/dataset.py [hunk 1]
```diff
@@ -48,3 +53,67 @@ def graph(self) -> object:
...
+    train_set: Optional[List[List[OnDiskTVTSet]]]
+    validation_set: Optional[List[List[OnDiskTVTSet]]]
+    test_set: Optional[List[List[OnDiskTVTSet]]]
+
+
+class OnDiskDataset(Dataset):
+    """An on-disk dataset.
+
+    An on-disk dataset is a dataset which stores the data on disk. It is
+    usually used when the dataset is too large to fit into the memory.
```
Comment (frozenbugs) [inline python/dgl/graphbolt/dataset.py:89]: This comment is a little bit confusing, what do you mean by the dataset is too large to fit into the memory? Before we run the code, everything should be on disk, and one we start training, some of the data will be load to memory, and some will remain on disk.

Can you clarify based on our offline discussion?

Comment (Rhett-Ying) [reply python/dgl/graphbolt/dataset.py:89]: sure, the comment here should be refined

File: tests/python/pytorch/graphbolt/test_dataset.py [hunk 2]
```diff
@@ -14,3 +18,36 @@ def test_Dataset():
         _ = dataset.graph()
     with pytest.raises(NotImplementedError):
         _ = dataset.feature()
+
+
+def test_OnDiskDataset_TVTSet():
```
Comment (frozenbugs) [inline tests/python/pytorch/graphbolt/test_dataset.py:23]: You plan to implement train_set e.t.c. in the future PR, right?
This actually does not test _TVTSet, just initialization and metadata validation.
```suggestion
def test_OnDiskDataset_Initialization():
```

**Review by frozenbugs — APPROVED** [review]

File: tests/python/pytorch/graphbolt/test_dataset.py (hunk 2 above)
Comment (Rhett-Ying) [reply tests/python/pytorch/graphbolt/test_dataset.py:23]: yes, I'd like to parse TVT in several PRs so I use the name `test_xxx_TVTSet()`. As for the suggested name `Inialization`, it indicates init the whole dataset for me. But actually, I'd like to test init TVT.

**Pull request merged**